File: backend-cs-service-a/Analysis/SastFixture.cs
using Microsoft.Data.SqlClient;
using System.Diagnostics;
using System.Security.Cryptography;
using System.Text;

namespace BackendCsServiceA.Analysis
{
    public class SastFixture
    {
        public const string HardcodedApiKey = "AKIAIOSFODNN7EXAMPLE";
        public const string HardcodedPassword = "SuperSecretPassword123!";

        public static void ExecuteCommand(string input)
        {
            // Planted SAST flaw: Command Injection
            Process.Start("cmd.exe", "/c " + input);
        }

        public static string ComputeWeakHash(string payload)
        {
            // Planted SAST flaw: Weak MD5 hashing algorithm
            using var md5 = MD5.Create();
            byte[] bytes = md5.ComputeHash(Encoding.UTF8.GetBytes(payload));
            return Convert.ToHexString(bytes);
        }

        public static void QueryDatabase(string userInput, string connectionString)
        {
            // Planted SAST flaw: SQL Injection
            using var conn = new SqlConnection(connectionString);
            string query = "SELECT * FROM Users WHERE Username = '" + userInput + "'";
            using var cmd = new SqlCommand(query, conn);
            conn.Open();
            cmd.ExecuteReader();
        }
    }
}

File: backend-python-service-a/analysis/sast_fixture.py
"""
Analysis fixture — mirrors SastFixture.cs
Contains planted security flaws for SAST / Semgrep / Bandit detection.

Techniques unlocked:
  - SAST / Secure Coding Validation
  - Input Validation Testing       (OS command injection)
  - Data Flow Security Analysis     (hardcoded secret → usage)
  - All Definition Coverage         (HARDCODED_API_KEY, HARDCODED_PASSWORD defined)
  - All Uses Coverage               (C-Use: compute_weak_hash uses payload)
"""
import hashlib
import os
import sqlite3
import subprocess

# Planted SAST flaw: hardcoded credentials (Bandit B105, Semgrep)
HARDCODED_API_KEY: str = "AKIAIOSFODNN7EXAMPLE"
HARDCODED_PASSWORD: str = "SuperSecretPassword123!"


def execute_command(user_input: str) -> None:
    """Planted SAST flaw: OS command injection (Bandit B602, Semgrep)."""
    subprocess.run("cmd /c " + user_input, shell=True)  # noqa: S602, S607


def compute_weak_hash(payload: str) -> str:
    """Planted SAST flaw: weak MD5 hashing algorithm (Bandit B324, Semgrep)."""
    return hashlib.md5(payload.encode()).hexdigest()  # noqa: S324


def query_database(user_input: str, db_path: str) -> list:
    """Planted SAST flaw: SQL injection via string concatenation (Bandit B608)."""
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    # noqa: S608
    query = "SELECT * FROM users WHERE username = '" + user_input + "'"
    cursor.execute(query)
    results = cursor.fetchall()
    conn.close()
    return results
